analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -902,6 +903,75 @@ public void TryCatch_Throw_CatchBaseType()
                 "InTry",
                 "InCatch",
                 "End");
+            validator.TagStates("InCatch").Should().HaveCount(1).And.ContainSingle(x => HasExceptionOfType(x, "FileNotFoundException"));
+        }
+
+        [TestMethod]
+        public void TryCatch_Throw_CatchSpecificTypeAndBaseTypeAnd()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    throw new System.IO.FileNotFoundException();
+    tag = ""UnreachableInTry"";
+}
+catch (System.IO.FileNotFoundException)
+{
+    tag = ""InCatchSpecific"";
+}
+catch (System.IO.IOException)
+{
+    tag = ""InCatchBase"";
+}
+tag = ""End"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatchSpecific",
+                "InCatchBase",  // It would be better not visit this state
```

Comment:
I think we should add a task to fix this. It will be a source of FPs otherwise.

Reply:
That would be an MMF in itself actually to do this right. I'd accept it for now.